src/Microsoft.OData.Core/UriParser/ExpressionLexer.cs

```diff
@@ -1201,14 +1203,15 @@ private void AdvanceThroughBalancedExpression(char startingCharacter, char endin
 
 
         /// <summary>Parses an identifier by advancing the current character.</summary>
-        private void ParseIdentifier()
+        /// <param name="includingDots">Optional flag for whether to include dots as part of the identifier.</param>
+        private void ParseIdentifier(bool includingDots = false)

```

Comment:
>bool includingDots = false [](start = 37, length = 26)

We pass this as true if the leading character is an "@", otherwise we pass this as false.  Rather than making this a parameter, should we just set includingDots = this.ch==UriQueryConstants.AnnotationPrefix within the function?

Reply:
It would be equivalent to the current logic, but explicitly passing in as bool from caller is causing less ripple, cleaner therefore less risky and more flexible in case this method is needed later in other place. Therefore I prefer explicitly passing in a bool value unless you have strong opinion on this.

---
In reply to: [176544495](https://github.com/OData/odata.net/pull/1123#discussion_r176544495) [](ancestors = 176544495)